parser/src/ast/operation.rs

```diff
@@ -1,6 +1,15 @@
+use enum_assoc::Assoc;
+use lexer::token::TokenType;
+
 use crate::ast::Expr;
+use crate::ast::operation::BinaryOperator::*;
+
+pub const ARITHMETICS: &[BinaryOperator] = &[Plus, Minus, Times, Divide, Modulo];
+pub const COMPARISONS: &[BinaryOperator] = &[EqualEqual, NotEqual, Less, LessEqual, Greater, GreaterEqual];
+pub const BOOLEANS: &[BinaryOperator] = &[And, Or];
+
 
-/// A binary operation between two expressions.
+/// An arithmetic operation between two expressions.
```

Comment:
```suggestion
/// A binary operation between two expressions.
```
`||` and `&&` are logic operations, not arithmetic.